gaming/src/golang/item-service/main.go

```diff
@@ -0,0 +1,182 @@
+		c.MustGet("spanner_client").(spanner.Client)
+}
+
+func createItem(c *gin.Context) {
+	var item models.GameItem
+
+	if err := c.BindJSON(&item); err != nil {
+		c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	ctx, client := getSpannerConnection(c)
+	err := item.Create(ctx, client)
+	if err != nil {
+		c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	c.IndentedJSON(http.StatusCreated, item.ItemUUID)
+}
+
+// TODO: used by game server to generate load. Should not be called by other entities,
+//  so restrictions should be implemented
+func getItemUUIDs(c *gin.Context) {
+	ctx, client := getSpannerConnection(c)
+
+	items, err := models.GetItemUUIDs(ctx, client)
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "No items exist"})
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, items)
+}
+
+func getItem(c *gin.Context) {
+	var itemUUID = c.Param("id")
+
+	ctx, client := getSpannerConnection(c)
+
+	item, err := models.GetItemByUUID(ctx, client, itemUUID)
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "item not found"})
+		return
+	}
+
+	type ReturnGameItem struct {
+		ItemUUID, Item_name, Item_value string
+		Available_time                  time.Time
+		duration                        int64
+	}
+
+	gi := ReturnGameItem{ItemUUID: item.ItemUUID, Item_name: item.Item_name, Item_value: item.Item_value.FloatString(2),
+		Available_time: item.Available_time, duration: item.Duration}
+
+	c.IndentedJSON(http.StatusOK, gi)
+}
+
+// Update a player balance with a provided amount. Result is a JSON object that contains PlayerUUID and AccountBalance
+func updatePlayerBalance(c *gin.Context) {
+	var player models.Player
+	var ledger models.PlayerLedger
+
+	if err := c.BindJSON(&ledger); err != nil {
+		c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	ctx, client := getSpannerConnection(c)
+	err := ledger.UpdateBalance(ctx, client, &player)
+	if err != nil {
+		c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	type PlayerBalance struct {
+		PlayerUUID, AccountBalance string
+	}
+
+	balance := PlayerBalance{PlayerUUID: player.PlayerUUID, AccountBalance: player.Account_balance.FloatString(2)}
+	c.IndentedJSON(http.StatusOK, balance)
+}
+
+func getPlayer(c *gin.Context) {
+	ctx, client := getSpannerConnection(c)
+	player, err := models.GetPlayer(ctx, client)
+	if err != nil {
+		c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, player)
+}
+
+func addPlayerItem(c *gin.Context) {
+	var playerItem models.PlayerItem
+
+	if err := c.BindJSON(&playerItem); err != nil {
+		c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	ctx, client := getSpannerConnection(c)
+	err := playerItem.Add(ctx, client)
+	if err != nil {
+		c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	c.IndentedJSON(http.StatusCreated, playerItem)
+}
+
+// func getPlayerItem(c *gin.Context) {
```

Comment:
nit: remove

Reply:
I was imagining leaving the skeleton code for hands-on users to implement this as a function, along with the endpoint.

But I think I can handle that outside of the repo, so I'll remove.